support/oidc-discovery-provider/cert_manager_test.go

```diff
@@ -0,0 +1,423 @@
+package main
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"math/big"
+	"reflect"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"github.com/sirupsen/logrus/hooks/test"
+	"github.com/spiffe/spire/pkg/common/x509util"
+	"github.com/spiffe/spire/test/testkey"
+	"github.com/stretchr/testify/require"
+)
+
+var logger, logHook = test.NewNullLogger()
+
+func TestTLSConfig(t *testing.T) {
+	oidcServerKey := testkey.MustEC256()
+	oidcServerKeyDer, err := x509.MarshalECPrivateKey(oidcServerKey)
+	require.NoError(t, err)
+
+	certTmpl := &x509.Certificate{
+		SerialNumber: big.NewInt(0),
+		NotAfter:     time.Now().Add(time.Hour),
+		Subject: pkix.Name{
+			Country:    []string{"BR"},
+			CommonName: "oidc-provider-discovery.example.com",
+		},
+	}
+	oidcServerCert, err := x509util.CreateCertificate(certTmpl, certTmpl, oidcServerKey.Public(), oidcServerKey)
+	require.NoError(t, err)
+	require.NotNilf(t, oidcServerCert, "oidcServerCert is nil")
+
+	oidcServerKeyPem := pem.EncodeToMemory(&pem.Block{
+		Type:  "PRIVATE KEY",
+		Bytes: oidcServerKeyDer,
+	})
+
+	oidcServerCertPem := pem.EncodeToMemory(&pem.Block{
+		Type:  "CERTIFICATE",
+		Bytes: oidcServerCert.Raw,
+	})
+
+	certTmpl.Subject.Country = []string{"AR"}
+	oidcServerCertUpdated1, err := x509util.CreateCertificate(certTmpl, certTmpl, oidcServerKey.Public(), oidcServerKey)
+	require.NoError(t, err)
+
+	oidcServerKeyNew := testkey.MustEC256()
+	oidcServerKeyNewDer, err := x509.MarshalECPrivateKey(oidcServerKeyNew)
+	require.NoError(t, err)
+
+	oidcServerCertUpdated2, err := x509util.CreateCertificate(certTmpl, certTmpl, oidcServerKeyNew.Public(), oidcServerKeyNew)
+	require.NoError(t, err)
+
+	certTmpl.Subject.Country = []string{"US"}
+
+	oidcServerCertUpdated3, err := x509util.CreateCertificate(certTmpl, certTmpl, oidcServerKey.Public(), oidcServerKeyNew)
+	require.NoError(t, err)
+
+	tmpDir := t.TempDir()
+
+	err = writeFile(tmpDir+keyFilePath, oidcServerKeyPem)
+	require.NoError(t, err)
```

Comment:
nit: these test helpers could take the *testing.T and do the require.NoError internally instead of returning an error that has to be checked.